### ISSUE TITLE
Expectations in kernel weights

### ISSUE THREAD

**Issue by martinfleis** [pysal/libpysal#790]
As I am playing with `gwlearn` and kernels over there, compared to `mgwr`, I noticed that kernels in libpysal behave unexpectedly (besides an actual bug in #789). What I would assume that we use the kernel function, say Gaussian, to assign weights from 1 at distance 0 progressively to 0. But that is not the case. The progression of a Gaussian kernel goes from 0.4 as shown in the fig below (bandwidth 100). The same happens in GeoDa.

![Image](https://github.com/user-attachments/assets/40fdfd51-645a-4a75-af29-4699a58ebe3a)

However, if I use the kernel as defined in `mgwr`, it behaves as I expected.

```py
def _gaussian_mgwr(distances, bandwidth):
    zs = distances / bandwidth 
    return numpy.exp(-0.5 * ((zs)**2))
```

![Image](https://github.com/user-attachments/assets/2f4d11d0-4dd1-4bea-b9e4-e23ee6be4c35)

The reason I don't understand why we should not start at 1 is that if I get a kernel-based W, and want to assign self weight afterwards, I am pretty much without options of doing that correctly. The value for 0 is 0.3989422804014327. Though I would intuitively expect it to be 1.

Similar behaviour, albeit not that drastic, is in a few of the other kernels.

The formula for Gaussian we use is this:

```py
def _gaussian(distances, bandwidth):
    u = distances / bandwidth
    return numpy.exp(-((u / 2) ** 2)) / (numpy.sqrt(2 * numpy.pi))
```

Why is it divided by `(numpy.sqrt(2 * numpy.pi))`? If we don't then it matches MGWR and also the Gaussian kernel definition in support vector machine and elsewhere.

Can someone shed some light on this? It follows the formula I've found in @sjsrey's _Modern spatial econometrics in practice_ but have no clue where that comes from and why it looks like this.

**Comment by martinfleis** [comment]
Just making some notes here, primarily for myself to wrap my head around this.

Over in https://github.com/uscuni/gwlearn/issues/3#issuecomment-2920584059 @TaylorOshan mentions that we need to take a square root of weights before passing onto sklearn. Which applies to the formula in MGWR. If we do sqrt of that, we get exactly the same curve as we would get in libpysal if we excluded the division.

```py
def _libpysal_no_division(distances, bandwidth):
    u = distances / bandwidth
    return numpy.exp(-((u / 2) ** 2))

def _mgwr(distances, bandwidth):
    zs = distances / bandwidth 
    return numpy.exp(-0.5 * ((zs)**2))

x = numpy.linspace(10, 500)
plt.plot(x, _libpysal_no_division(x, 100), label="lib_no_division")
plt.plot(x, numpy.sqrt(_mgwr(x, 100)), label="sqrt of mgwr", linestyle="--")
plt.plot(x, _mgwr(x, 100), label="mgwr")
plt.legend()
```

![Image](https://github.com/user-attachments/assets/c7191d56-17a6-44d4-bccd-8ce3474a6ee8)

**Comment by martinfleis** [comment]
I think I got that far in my understanding. What I don't get is why would I want the normalised option. 

Shall we maybe add a keyword to build_kernel controlling this normalisation?

**Comment by ljwolf** [comment]
There are different ways of normalizing kernels. Some normalize the kernel values so that K(0)=1. Others normalize the kernel so that `integral(K(X))=1` over the domain of the kernel. Sometimes, we call the latter "proper" kernels in Bayesian computation, since they're "proper" probability distributions. See, for instance, [this list.](https://en.wikipedia.org/wiki/Kernel_(statistics)#Kernel_functions_in_common_use). If we were opinionated, we'd standardize the kernels so that the integral is always 1, or that K(0)=1.

**Comment by sjsrey** [comment]
Adding a keyword arg would increase the clarity and allow for different use cases (i.e., kernels as proper pdfs, versus normalized kernels).

**Comment by martinfleis** [comment]
Agreed to schedule a sprint on this.
